proxy/src/console/provider/neon.rs

```diff
@@ -328,19 +330,24 @@ async fn parse_body<T: for<'a> serde::Deserialize<'a>>(
         info!("request succeeded, processing the body");
         return Ok(response.json().await?);
     }
+    let s = response.bytes().await?;
+    // Log plaintext to be able to detect, whether there are some cases not covered by the error struct.
+    info!("response_error plaintext: {:?}", s);
 
     // Don't throw an error here because it's not as important
     // as the fact that the request itself has failed.
-    let body = response.json().await.unwrap_or_else(|e| {
+    let mut body = serde_json::from_str(&s).unwrap_or_else(|e| {
```

Comment:
Looks like this needs to change
```suggestion
    let mut body = serde_json::from_slice(&s).unwrap_or_else(|e| {
```